frontend/packages/dev-console/src/utils/pipeline-actions.tsx

```diff
@@ -31,28 +26,30 @@ export const getPipelineRunData = (pipeline: Pipeline, latestRun?: PipelineRun):
     resourceRef: resource.resourceRef,
   }));
 
+  const pipelineName = pipeline ? pipeline.metadata.name : latestRun.spec.pipelineRef.name;
```

Comment:
The signature of this method says `latestRun` can be optional (aka `undefined` / omitted). The conditional check you removed at the start of the method makes this a potential fail-statement if `pipeline` is provided as `null` without a `latestRun`. We really should have tests for this function, and one of those tests should be to express the fail states.

Please put in a similar fallback to what you removed to cover the missing data use case.
Please also either add tests for this function or log a JIRA Task to write tests next week (after feature freeze).

Reply:
Also make ` pipeline` type also a null or undefined. Since you call this function now with a `null` pipeline.

Reply:
Modified type & added parameter check. Also added tests.